src/pages/homepage/homepage.test.js

```diff
@@ -0,0 +1,47 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { BrowserRouter } from "react-router-dom";
+import Homepage from ".";
+
+const productService = require("../../services/productService");
+
+const products = [
+  { name: "something", id: 1, image: null, price: 1000 },
+  { name: "chips", id: 2, image: "", price: 30 },
+  { name: "chocolate", id: 3, image: null, price: 50 },
+];
+
+const renderHomepage = () => {
+  render(
+    <BrowserRouter>
+      <Homepage />
+    </BrowserRouter>
+  );
+};
+
+describe("Homepage", () => {
+  beforeEach(() => {
+    jest.spyOn(productService, "getProducts").mockResolvedValue(products);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
```

Comment:
This logic can be moved into `beforeAll` and we don't need to restore mocks at least in the current scenario.

```suggestion
  beforeAll(() => {
    jest.spyOn(productService, "getProducts").mockResolvedValue(products);
  });
```